src/wasm/literal.cpp

```diff
@@ -71,8 +71,9 @@ Literal::Literal(const uint8_t init[16]) : type(Type::v128) {
 
 Literal::Literal(std::shared_ptr<GCData> gcData, HeapType type)
   : gcData(gcData), type(type, NonNullable) {
-  // The type must be a proper type for GC data.
-  assert((isData() && gcData) || (type.isBottom() && !gcData));
+  // The type must be a proper type for GC data: either a struct, array, or
+  // string; or a null.
+  assert(((isData() || isString()) && gcData) || (type.isBottom() && !gcData));
```

Comment:
Hmm, we might want to consider including strings in `isData` since there's no obvious reason to exclude them. That might break some assumptions, though, so I wouldn't want to do it in this PR. Another option would be to remove `isData` entirely now that there is no `data` type.

Reply:
I agree we probably shouldn't do anything in this PR, as you said. Makes sense to consider both of the options you suggested later.

Reply:
But it looks like we already are including strings in `Type::isData()` down in wasm-type.cpp, so the current state seems inconsistent. Am I missing something?

Reply:
Ah, now I see what you mean, sorry for missing it before... I didn't put your comments together in my head.

Yeah, good point, this is inconsistent. Maybe the simplest thing is to go all-in with strings as data. That has a simple logic: they are backed by heap-allocated data, like structs and arrays.

Reply:
Done.